### PR TITLE
Getting local requests to work with build_and_redeploy.sh

### PR DESCRIPTION
**Pull request by andresilva91**
## Motivation

Requests to a local Kubernetes cluster were not working

## Proposal

This PR fixes that and introduces some extra functionality:
* Removing `kube` feature
* Creating a couldbuild YAML config file so that we can run Cloud Builds based on different Dockerfiles
* Added `--cloud-build` option to `build_and_redeploy.sh` and `redeploy.sh`. This will make it possible to do a cloud build instead of a local docker build. Cloud builds tend to be faster.
* Added `--port-forward` option to `build_and_redeploy.sh` and `redeploy.sh`. With this you can just run the command and once it finishes and starts port forwarding, you know your local validator is ready to go and start receiving requests from clients.

Ideally this would've been multiple PRs, but going with a big one for now.

## Test Plan

Ran the scripts with the different modes, sent requests to the local validator and saw them succeed.

## Links

This PR is rebased on top of #1015, so it contains the changes from it.

## Release Plan

- [x] All good!
- [ ] Need to bump the major/minor version number in the next release of the crates.
- [ ] Need to update the developer manual.
- [ ] This PR is adding or removing Cargo features.
- [ ] Release is blocked and/or tracked by other issues (see links above)

## Reviewer Checklist

- [ ] The title and the summary of the PR are short and descriptive.
- [ ] The proposed solution achieves the goals stated in the PR.
- [ ] The test plan is reproducible and provides sufficient coverage.
- [ ] The release plan is adequate.
- [ ] The commits correspond to distinct logical changes.
- [ ] The code follows the [coding guidelines](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md).
- [ ] The proposed changes look correct.
- [ ] The CI is passing.
- [ ] All of the above!

### CODE REVIEW

File: kubernetes/linera-validator/redeploy.sh [hunk 1]
```diff
@@ -1,11 +1,81 @@
...
+    docker rm linera-test-local;
+    docker run -d --name linera-test-local linera-test:latest && docker cp linera-test-local:/opt/linera/wallet.json /tmp/
+fi
+
+echo "Make sure the terminal you'll run the linera client from has these exports:"
+echo 'export LINERA_WALLET=/tmp/wallet.json'
+echo 'export LINERA_STORAGE="rocksdb:/tmp/linera.db"'
+
+export LINERA_WALLET=/tmp/wallet.json
+export LINERA_STORAGE="rocksdb:/tmp/linera.db"
```
Comment (andresilva91) [inline kubernetes/linera-validator/redeploy.sh:72]: One thing we might consider doing here is deleting this DB file as part of this script. I've seen issues where it tries to connect to an old DB folder and things fail

Comment (ma2bd) [reply kubernetes/linera-validator/redeploy.sh:72]: Yes we have a task to refactor storage options

**Comment by christos-h** [comment]
So it turns out #1015 works - but I'll close it in favour of this as this scripts are more developed.

BTW - since we are not using calling Kube APIs from Rust code, shouldn't the dependency go as well?

**Comment by andresilva91** [comment]
Yup, good catch! Missed those

**Review by christos-h — APPROVED** [review]
LGTM just make sure to remove the Grafana / Prometheus commits and we'll do those separately.

**Pull request merged**